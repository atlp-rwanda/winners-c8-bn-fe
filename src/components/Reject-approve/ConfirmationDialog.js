import React from 'react';
import './confirmDialogue.scss'

function ConfirmationDialogue({
  dialogueStatus,
  handleConfirm,
  handleCancel,

}) {
  let status;
  dialogueStatus !== ('approved' || 'rejected')
    ? (status = dialogueStatus)
    : dialogueStatus === 'approved'
    ? (status = 'approve')
    : (status = 'reject');

  return (
    <>
      <div className="container-dialog">
        <span
          data-testid="confirmDialogue_text"
          className="mb-5 capitalize"
        >{`Do you really want to ${status} trip request?`}</span>
        <div className="button-container">
          <button className="cancel-button" onClick={() => handleCancel(false)}>
            NO
          </button>
          <button
            className="confirmation-button"
            onClick={() => handleConfirm(dialogueStatus)}
          >
            Yes
          </button>
        </div>
      </div>
     
    </>
  );
}

export default ConfirmationDialogue;
