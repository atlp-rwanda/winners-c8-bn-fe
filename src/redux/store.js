// this file will configure the redux's store
import { createStore } from "redux";
import rootReducer from "./reducers";

let authToken = ""; 
if(window.localStorage.getItem("auth-token")) {
  authToken = window.localStorage.getItem("auth-token");
}

const initialState = {
    auth: {token: authToken},
    navbar: {currentPage: "HOME"},
};
const store = createStore(
    rootReducer,
    initialState // sets initial state
  );

export default store;
