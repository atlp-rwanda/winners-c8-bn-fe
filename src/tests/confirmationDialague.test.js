import React from 'react';
import { Provider } from 'react-redux';
import ConfirmationDialog from '../components/Reject-approve/ConfirmationDialog';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/store';
// import {render} from '@testing-library/jest-dom'
describe.only('Dialogue',() =>{it(' <ConfirmationDialog /> matches snapshot', () => {
  const component = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <ConfirmationDialog />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
} )
