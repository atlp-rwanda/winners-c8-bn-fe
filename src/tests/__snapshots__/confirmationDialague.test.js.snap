// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialogue  <ConfirmationDialog /> matches snapshot 1`] = `
<div
  className="container-dialog"
>
  <span
    className="mb-5 capitalize"
    data-testid="confirmDialogue_text"
  >
    Do you really want to undefined trip request?
  </span>
  <div
    className="button-container"
  >
    <button
      className="cancel-button"
      onClick={[Function]}
    >
      NO
    </button>
    <button
      className="confirmation-button"
      onClick={[Function]}
    >
      Yes
    </button>
  </div>
</div>
`;
